src/main/java/strategy/OperationStrategyImpl.java

```diff
@@ -0,0 +1,25 @@
+package strategy;
+
+import handler.OperationHandler;
+import java.util.Map;
+import model.FruitTransaction;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> handlers;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation, OperationHandler> handlers) {
+        this.handlers = handlers;
+    }
+
+    @Override
+    public void handleOperation(FruitTransaction transaction) {
+        OperationHandler handler = handlers.get(transaction.getOperation());
+        if (handler != null) {
+            handler.handle(transaction);
+        } else {
+            throw new RuntimeException("No handler found for operation: "
+                    + transaction.getOperation());
+        }

```

Comment:
in cases like this, invert if condition, throw exception early, you wont'need else after return/throw, and code will be easier to read
https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement